projects/ancestor/ancestor.py

```diff
@@ -1,3 +1,111 @@
+class Stack():
+    def __init__(self):
+        self.stack = []
+
+    def push(self, value):
+        self.stack.append(value)
+
+    def pop(self):
+        if self.size() > 0:
+            return self.stack.pop()
+
+        raise IndexError("cannot pop empty stack")
+
+    def size(self):
+        return len(self.stack)
+
+
+class Graph():
+    def __init__(self):
+        self.vertices = {}
+
+    def add_vertex(self, value):
+        self.vertices[value] = set()
+
+    def isvertex(self, value):
+        return value in self.vertices
+
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].add(v2)
+            return
+
+        raise IndexError("verts not found")
```

Comment:
I think you could put this in an else block and then you don't need the `return` on line 31. But it's up to you 👍